python-modules/cis_crypto/cis_crypto/operation.py

```diff
@@ -73,20 +82,63 @@ def _get_public_key(self):
                     )
                 )
             )
-            key_name = self.config('public_key_name', namespace='cis')
+            key_name = self.config('public_key_name', namespace='cis', default='access-file-key')
             file_name = '{}.pub.pem'.format(key_name)
             fh = open((os.path.join(key_dir, file_name)), 'rb')
             key_content = fh.read()
             key_construct = jwk.construct(key_content, 'RS256')
             return key_construct.to_dict()
-        elif self.well_known_mode is 'http':
+        elif self.well_known_mode == 'http' or self.well_known_mode == 'https':
             # Go get it from the .well-known endpoint and load as json
             # return a dictionary of the json loaded data
-            pass
+            well_known_url = self.config('well_known_url', namespace='cis')
+            res = requests.get(well_known_url)
+
+            if res.status_code == 200:
+                key_list = self._reduce_keys(res.json())
+                return key_list
+
+    def _reduce_keys(self, well_known_response):
+        access_file_keys = well_known_response['access_file']['jwks_keys']
+        publisher_keys = well_known_response['publishers_supported']
+
+        keys = []
+
+        for key in access_file_keys:
```

Comment:
Wait, so why iterate over the content of `access_file_keys` if you don't use the content. Could you instead do
```
for i in range(len(access_file_keys)):
    keys.append(
        {
            'access_file_key_{}'.format(i): access_file_keys.get(str(i))
        }
    )
```